### PR TITLE
Align pandas objects with named indexes

### PR DESCRIPTION
**Pull request by MaximilianR**
closes https://github.com/xray/xray/issues/664. 
What's new & docs forthcoming

### CODE REVIEW

File: xray/core/alignment.py [hunk 1]
```diff
@@ -103,9 +103,17 @@ def partial_align(*objects, **kwargs):
...
     """Align all DataArrays in the provided dict, leaving other values alone.
     """
-    alignable = [k for k, v in variables.items() if hasattr(v, 'indexes')]
-    aligned = align(*[variables[a] for a in alignable], join=join, copy=copy)
+    from .dataarray import DataArray
+    from pandas.core.generic import NDFrame
+
     new_variables = OrderedDict(variables)
+    # if an item is a Series / DataFrame / Panel, try and wrap it in a constructor
+    new_variables.update(OrderedDict(
```
Comment (shoyer) [inline xray/core/alignment.py]: I think you can just pass a list of tuples here rathe than using an OrderedDict

File: xray/core/alignment.py [hunk 2]
```diff
@@ -103,9 +103,17 @@ def partial_align(*objects, **kwargs):
...
+        (k, DataArray(v)) for k, v in variables.items() if isinstance(v, NDFrame)
```
Comment (shoyer) [inline xray/core/alignment.py]: I would rather use `(pd.Series, pd.DataFrame, pd.Panel)` in place of `NDFrame`, which I'm not sure is part of the public API.

File: xray/test/test_dataset.py [hunk 3]
```diff
@@ -199,6 +199,22 @@ def test_constructor_auto_align(self):
         with self.assertRaisesRegexp(ValueError, 'conflicting sizes'):
             Dataset({'a': a, 'b': b, 'e': e})
 
+    def test_constructor_pandas(self):
+
+        ds = self.make_example_math_dataset()
+        pandas_objs = OrderedDict(
+            (var_name, ds[var_name].to_pandas()) for var_name in ['foo','bar']
+        )
+        ds_based_on_pandas = Dataset(variables=pandas_objs, coords=ds.coords, attrs=ds.attrs)
```
Comment (shoyer) [inline xray/test/test_dataset.py]: First argument should be `data_vars` now :)

**Comment by shoyer** [comment]
Look very nice to me!

**Comment by MaximilianR** [comment]
@shoyer updated, docs & what's new added, and green

**Comment by shoyer** [comment]
Thanks!

**Pull request merged**